File: programming files/remove_duplicates.py
#Bugs introduced+OD

def remove_duplicates(arr):
    # sorts array
    arr.sort()
    unique_arr = []
    removed_elements = []
    # iterates through the array
    for i in range(len(arr)):
        # if i is 0 or not equal to the last element, append to unique array
        if i == 0 or arr[i] != arr[i-1]:
            unique_arr.append(arr[i])
        else:
            # else, append the element to the removed_elements list
            removed_elements.append(arr[i])
    return unique_arr, removed_elements

# Test the function
arr = [1, 2, 2, 3, 4, 4, 5]
unique_arr, removed_elements = remove_duplicates(arr)
print("Original array:", arr)
print("Unique array:", unique_arr)
print("Removed elements:", removed_elements)

File: programming files/remove_duplicates.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class remove_duplicates {

    public static List<Integer> removeDuplicates(List<Integer> arr, List<Integer> removedElements) {
        List<Integer> uniqueArr = new ArrayList<>();
        for (int i = 0; i < arr.size() - 1; i++) {
            // adds integer from list and adds to new list if it is not already added
            if (!uniqueArr.contains(arr.get(i + 1))) {
                uniqueArr.add(arr.get(i));
            } else {
                // if the uniqueArr contains the integer, add it to the removedElements list
                removedElements.add(arr.get(i));
            }
        }
        return uniqueArr;
    }

    public static void main(String[] args) {
        // Test the function
        List<Integer> arr = Arrays.asList(1, 2, 2, 3, 4, 4, 5);
        List<Integer> removedElements = new ArrayList<>();
        List<Integer> uniqueArr = removeDuplicates(arr, removedElements);
        System.out.println("Original array: " + arr);
        System.out.println("Unique array: " + uniqueArr);
        System.out.println("Removed elements: " + removedElements);
    }
}
